resource.go

```diff
@@ -60,13 +82,79 @@ func getResourceField(v interface{}) (f *structs.Field, err error) {
 	return
 }
 
+func getIdFromMap(m map[string]interface{}) (ret int, err error) {
```

Comment:
This function never returns `err` in any case, it's nice that you can at line 105 below just `return`, but I think it'd be clearer to remove the `err` from here and change that line.